### PR TITLE
feat: Allow parameter objects to be used for in mixins

### PR DESCRIPTION
**Pull request by Mitcheljager**
I had a mixin that looked like this:
```
@mixin projectileHeroSettings(hero, speed = 50, size = 0.1, radius = 0, timeBetweenShots = 0, maxRange = 0, delay = 0) {
```
The problem was that include result in a whole bunch of `50, 0.1, 0, 0, 0, 0.1` values because I only wanted to change the last value.

This PR changes that by allows parameter objects to be used in mixins, which lucky was a lot easier than expected, however it did present one issue; you might want to use parameter objects as values for your parameters, and how do we know if it's an object for the mixin or to be used literally? We don't. The solution I came up with is that if you provide only one parameters which is an object, we assume it's for the mixin. If you provide multiple we assume it's to be used literally.

That might be confusing to read so two examples:
```js
@mixin someMixin(one = 1, two = 2) {
  Some Action(Mixin.one);
  Some Action(Mixin.two);
}

@include someMixin({ one: 2, two: 3 })

// Results in
Some Action(2);
Some Action(3);
```

```js
@mixin someMixin(one = 1, two = 2) {
  Mixin.one;
  Mixin.two;
}

@include someMixin({ one: 2, two: 3 }, { one: 5, two: 6 })

// Results in
Some Action({ one: 2, two: 3 });
Some Action({ one: 5, two: 6 });
```

This has the downside that you can't have only one param that is an object, but this is easily remedied by including an empty param `@include someMixin({ one: 2 }, "");`. That's not great, but it's a way around an edge case.

### CODE REVIEW

File: spec/javascript/utils/compiler/mixins.test.js [hunk 1]
```diff
@@ -215,6 +215,36 @@ describe("mixins.js", () => {
       expect(result.fullMixin).toEqual(joinedItems)
       expect(disregardWhitespace(result.replaceWith)).toEqual(disregardWhitespace("Global.slot = 1;"))
     })
+
+    it("Should replace Mixin variables with parameter objects if it's given as the only param", () => {
```
Comment (netux) [inline spec/javascript/utils/compiler/mixins.test.js:219]: I think maybe this could be a warning on the linter?

Comment (Mitcheljager) [reply spec/javascript/utils/compiler/mixins.test.js:219]: Hmm maybe, but it is intended, not a warning. It's difficult to convey though
